// Copyright 2018 REMME
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ------------------------------------------------------------------------

syntax = "proto3";


message PubKeyMethod {
    enum Method {
        STORE = 0;
        REVOKE = 1;
    }
}

message NewPubKeyPayload {
    enum EntityType {
        PERSONAL = 0;
        SERVER = 1;
    }

    enum PubKeyType {
        RSA = 0;
    }
    // PEM Encoded public key
    string public_key = 1;
    PubKeyType public_key_type = 2;
    // For future ACME support(Now only Personal)
    EntityType entity_type = 3;
    // hash of PEM encoded public key of certificate
    string entity_hash = 4;
    string entity_hash_signature = 5;
    // UNIX time when validity begins
    uint32 valid_from = 6;
    // UNIX time when validity ends
    uint32 valid_to = 7;
}

message RevokePubKeyPayload {
    // The address of a certificate in the storage generated by the transaction processor.
    string address = 1;
}

message PubKeyStorage {
    // The UID field of a certificate.
    string owner = 1;
    // indicates if this key was revoked
    NewPubKeyPayload payload = 2;
    bool revoked = 3;
}
